code/modules/mob/living/carbon/human/species_types/squidpeople.dm

```diff
@@ -32,6 +46,25 @@
         . = capitalize(squid_name(genderToFind))
         if(!findname(.))
             break
-			
+
 /datum/species/squid/after_equip_job(datum/job/J, mob/living/carbon/human/H)
-	H.grant_language(/datum/language/rlyehian)	
+	H.grant_language(/datum/language/rlyehian)
+
+/datum/species/squid/spec_updatehealth(mob/living/carbon/human/H)
+	.=..()
+	if(H.stat != DEAD )
+		fixed_mut_color = copytext(current_color, 2)
+	else
+		fixed_mut_color = rgb(128,128,128)
+	H.update_body()
+
+/datum/action/innate/squid_change
+	name = "Color Change"
+	check_flags = AB_CHECK_CONSCIOUS
+	icon_icon = 'icons/mob/animal.dmi'
+	button_icon_state = "squid"
+
+/datum/action/innate/squid_change/Activate()
```

Comment:
I'm not sure if actions have built-in spam prevention, so test if it does or not and I'll show you how to add it if it doesn't.

Reply:
Either way it may be worth putting a 5 second delay between uses, tbh.

Reply:
I agree, I tried briefly (before getting burned out) finding the code but couldn't find any type of cooldown for innate abilities like this